### PR TITLE
Madison/collection permissioning

### PR DESCRIPTION
**Pull request by madisonmay**
Reviewers: 
----------

- [x] @sihrc 
- [x] @anneCarlson 

What's the purpose of the PR?
-----------------------------

Add support for custom collection permissioning system.

Description of changes
----------------------

Addition of register, deregister, rename, authorize, and deauthorize endpoints

Notes to reviewers
------------------

Some of the docs strings were also modified when I noticed they were incorrect.  Probably should have handled those changes in a separate PR.
Along the same lines, I also removed the "private" keyword which is passed into custom API methods.  It's not actually used anywhere.

Keep in mind while reviewing code:
----------------------------------
- Is relevant code tested?
- Are added functions/methods documented?
- Separation of concerns (SOC)
- Don't repeat yourself (DRY) 
- Limit the number of positional arguments
- Is function/method length reasonable?
- Can code be broken down into smaller components?
- Where do added functions/methods belong?
- Are variable names descriptive?
- Are errors handled appropriately?
- Can logic be simplified?
- Does the code make sense in context? (expand the diff)

### CODE REVIEW

File: indicoio/custom/custom.py [hunk 1]
```diff
@@ -204,6 +161,102 @@ def wait(self, interval=1):
...
+          to the appropriate destination.
+        cloud (optional) - String: Your private cloud domain, required only if the key has not been declared
+          elsewhere. This allows the API to recognize a request as yours and automatically route it
+          to the appropriate destination.
+        make_public (optional) - Boolean: When True, this option gives all indico users read access to your model.
+        """
+        kwargs['collection'] = self.collection
+        kwargs['make_public'] = make_public
+        url_params = {"batch": False, "api_key": api_key, "version": version, "method": "register"}
+        return api_handler(None, cloud=cloud, api="custom", url_params=url_params, private=True, **kwargs)
```
Comment (anneCarlson) [inline indicoio/custom/custom.py]: What do we want the behavior to be when someone tries to register a public collection?

File: indicoio/custom/custom.py [hunk 2]
```diff
@@ -204,6 +161,102 @@ def wait(self, interval=1):
...
+          to the appropriate destination.
+        cloud (optional) - String: Your private cloud domain, required only if the key has not been declared
+          elsewhere. This allows the API to recognize a request as yours and automatically route it
+          to the appropriate destination.
+        """
+        kwargs['collection'] = self.collection
+        kwargs['name'] = name
+        self.collection = name
+        url_params = {"batch": False, "api_key": api_key, "version": version, "method": "rename"}
+        return api_handler(None, cloud=cloud, api="custom", url_params=url_params, private=True, **kwargs)
```
Comment (anneCarlson) [inline indicoio/custom/custom.py]: I feel like the server should assume that `private=True` for each of these. Isn't it always the case?

File: indicoio/custom/custom.py (hunk 1 above)
Comment (madisonmay) [inline indicoio/custom/custom.py]: Error message saying the name is taken returned from the server.

File: indicoio/custom/custom.py (hunk 2 above)
Comment (madisonmay) [inline indicoio/custom/custom.py]: I think the private keyword should be here at all.  I don't think it's ever used and is probably an artifact from an old PR (given that I was working from the template of an existing custom collections method)

Comment (madisonmay) [inline indicoio/custom/custom.py]: Just checked the server, the private keyword is never used.

Comment (madisonmay) [inline indicoio/custom/custom.py]: Newest commit removes use of the private keyword argument entirely

**Pull request merged**